cmd/kube-proxy/app/server_others.go

```diff
@@ -317,6 +347,76 @@ func newProxyServer(
 	}, nil
 }
 
+func getDetectLocalMode(config *proxyconfigapi.KubeProxyConfiguration) proxyconfigapi.LocalMode {
+	switch config.DetectLocalMode {
+	case proxyconfigapi.LocalModeClusterCIDR:
+		return config.DetectLocalMode
+	default:
+		klog.V(4).Info("Defaulting detect local mode to ", string(proxyconfigapi.LocalModeClusterCIDR))
+		return proxyconfigapi.LocalModeClusterCIDR
+	}
+}
+
+func getIPTablesDetectLocal(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, ipt utiliptables.Interface) (detectLocal utiliptables.DetectLocalTraffic, err error) {
+	detectLocal = utiliptables.NewNoOpDetectLocal()
```

Comment:
I think this is easier to follow if you make this part of the `default` case for the switch statement below.

Reply:
actually it's not the default case. For example, it also applies when user selected LocalModeClusterCIDR, but did not set a ClusterCIDR. In that case too, we fallback to NoOpDetectLocal. I was thinking default would complicate the flow further?